SwiftIOSExperiments/Modules/Login/Contract/LoginContract.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  LoginContract.swift
+//  SwiftIOSExperiments
+//
+//  Created by James Santos on 14/2/17.
+//  Copyright © 2017 James Santos. All rights reserved.
+//
+
+import UIKit
+
+protocol LoginValidation {}
+
+protocol LoadingAuthentication {
+	var spinner: UIActivityIndicatorView { get }
```

Comment:
It's more "Loadable" than "Loading" for the nomenclature.

not mandatory for this sample application but for bnp one you will probably have a protocol Loadable mutualized and you will can inherited it for use-case protocol.